model_one/cli.py

```diff
@@ -1,198 +1,274 @@
-        self.status()
-        if self._status == 'Created':
+    @property
+    def is_initied(self) -> bool:
+        return self._id is not None
+    
+    @property
+    @inited
+    def is_ready(self):
+        return self.status is BeyondmlModelStatus.READY
+    
+    @property
+    @inited
+    def status(self) -> BeyondmlModelStatus:
+        r = BeyondmlAPI.status(self._id)
+        self._status = status = r["status"]
+        return BeyondmlModelStatus(status.lower())
+
+    @property
+    @inited
+    def type(self) -> BeyondmlModelType:
+        return BeyondmlModelType(self._model_type.lower())
+
+    @inited
+    def fit(
+        self,
+        train_X: Union[list, Series, None], 
+        train_y: Union[list, Series, None],
+        validate_X: Union[list, Series, None],
+        validate_y: Union[list, Series, None]
+    ):
+        if all(train_X, train_y, validate_X, validate_y):
             self.upload(train_X, train_y, validate_X, validate_y)
-
-        if self._status == 'DatasetsLoaded':
-            self.fit()
-        return self
-
-
-def load(filename):
-    with open(filename, 'r') as fl:
-        return BeyondmlModel(json.load(fl))
-
-
-def _create(data):
-    headers = _get_auth_header()
-    headers['Content-Type'] = 'application/json'
-    r = requests.post(_models_api(), headers=headers, data=json.dumps(data))
-    if r.status_code != 200:
-        raise Exception(r.text)
-    return BeyondmlModel(r.json())
-
-
-def _create_or_load(data: dict, filename: str = None):
-    if filename is not None and os.path.isfile(filename):
-        model = load(filename)
-        if model._model_type != data['model_type']:
-            raise Exception(
-                'Model in the file is not {}'.format(data['model_type']))
-        return model
-    m = _create(data)
-    m.save(filename)
-    return m
-
-def create_generator(filename: str = None, train_iters : int = None):
-    model = {'model_type': 'generator'}
-    if train_iters:
-        model.update({'model_params': {'train_iters': train_iters}})
-    return _create_or_load(model, filename)
-
-def create_classifier(filename: str = None, train_iters : int = None):
-    model = {'model_type': 'classifier'}
-    if train_iters:
-        model.update({'model_params': {'train_iters': train_iters}})
-    return _create_or_load(model, filename)
-
-
-def models():
-    r = requests.get(_models_api(), headers=_get_auth_header())
-    if r.status_code != 200:
-        raise Exception(r.text)
-    res = r.json()
-    res = []
-    if 'models' in res:
-        for k in res['models']:
-            res.append(BeyondmlModel(k))
-    return res
-
-
-def get_model(id):
-    r = requests.get('{}/{}/status'.format(_models_api(), id),
-                     headers=_get_auth_header())
-    if r.status_code != 200:
-        raise Exception(r.text)
-    return BeyondmlModel(r.json())
-
-
-def train_generator(filename: str, train_X, train_y, validate_X, validate_y, train_iters : int = None):
```

Comment:
I dont really get how these methods were supposed to be used
They are literally a shortcut for writing one line instead of two and as a benefit you can't recover id of a trained model
And they also snowball arguments of underlying functions